### ISSUE TITLE
Support for SnakeYaml 2.0

### ISSUE THREAD

**Issue by guglu9** [spring-cloud/spring-cloud-kubernetes#1302]
**SnakeYaml 2.0 is not supported**
io.kubernetes v18.0.0 has added support for SnakeYaml 2.0, can we upgrade to v18.0.0 please

**Sample Output**
Caused by: java.lang.NoSuchMethodError: org.yaml.snakeyaml.constructor.SafeConstructor: method 'void <init>()' not found
		at io.kubernetes.client.util.KubeConfig.loadKubeConfig(KubeConfig.java:83)
		at io.kubernetes.client.util.ClientBuilder.getClientBuilder(ClientBuilder.java:124)
		at io.kubernetes.client.util.ClientBuilder.standard(ClientBuilder.java:108)
		at io.kubernetes.client.util.ClientBuilder.standard(ClientBuilder.java:100)
		at io.kubernetes.client.util.ClientBuilder.defaultClient(ClientBuilder.java:82)
		at org.springframework.cloud.kubernetes.client.KubernetesClientUtils.kubernetesApiClient(KubernetesClientUtils.java:51)
		at org.springframework.cloud.kubernetes.client.KubernetesClientAutoConfiguration.apiClient(KubernetesClientAutoConfiguration

**Comment by guglu9** [comment]
```
<dependency>
      <groupId>org.springframework.cloud</groupId>
      <artifactId>spring-cloud-starter-kubernetes-fabric8-config</artifactId>
      <version>2.1.5</version>
      <scope>compile</scope>
    </dependency>
    <dependency>
      <groupId>org.springframework.cloud</groupId>
      <artifactId>spring-cloud-starter-kubernetes-client-config</artifactId>
      <version>2.1.5</version>
      <scope>compile</scope>
    </dependency>
    <dependency>
```

**Comment by ryanjbaxter** [comment]
I assume you are including snakeyaml 2.0 as well?

In fact the version of Spring Cloud Kubernetes you are using is using using version 13 of the Kubernetes Java client

**Comment by guglu9** [comment]
Yes, that's correct. As a temporary fix I've overridden `<kubernetes-native-client.version>18.0.0</kubernetes-native-client.version>` and it's working.

**Comment by ryanjbaxter** [comment]
OK, thats the best we can offer at the moment.  We cannot upgrade dependencies to new versions without potentially breaking others.

**Comment by guglu9** [comment]
Will this be available as part of 2.1.x once your PR gets merged?

**Comment by ryanjbaxter** [comment]
The PR is merged, they just need to do a release of the 17.x branch.  However the 2.1.x branch uses the 13.x branch of the kubernetes java client and they informed me there will not be a release of that branch. 

If the Kubernetes Java Client does a release of the 17.x branch we will pick that up in Spring Cloud Kubernetes 3.x

**Comment by edmundham** [comment]
Hi, thank you for the fix. However I'd like to confirm if this will not be shipped into 2.1.x. 
Spring Boot 2.7.10 has added support for SnakeYaml 2.0: https://github.com/spring-projects/spring-boot/releases/tag/v2.7.10#:~:text=Loading%20application.yml%20fails%20with%20NoSuchMethodError%20when%20using%20SnakeYAML%202.0 , but Spring-Cloud-Kubernetes 3.x requires Spring Boot 3.x, correct?

So I believe that means we have to use Spring Boot 3.x in order to use SnakeYaml 2.0 if we use spring-cloud-kubernetes?

**Comment by ryanjbaxter** [comment]
Spring Boot 2.7.10 is on COMPATIBLE with SnakeYaml 2.0, it does not use SnakeYaml 2.0.
https://github.com/spring-projects/spring-boot/issues/34405#issuecomment-1451004370

You still need to override the dependency version of SnakeYaml in your project if you want to use it in Spring Boot 2.7.10.

So you can still use Spring Cloud K8S 2.1.x, and Spring Boot 2.7.10 but I am not sure if the Kubernetes Java Client we are using (13.x) is compatible with SnakeYaml 2.0.0 and there is nothing we can do about that.

> So I believe that means we have to use Spring Boot 3.x in order to use SnakeYaml 2.0 if we use spring-cloud-kubernetes, because it will be shipped into spring-cloud-kubernetes 3.x only?

This would be the best known path to support SnakeYaml 2.0
